src/domain/events/Dispatcher.ts

```diff
@@ -0,0 +1,23 @@
+import { Event } from './Event'
+
+export type AppEventListener = (event: Event) => void
+
+export class EventDispatcher {
+  private listeners: AppEventListener[]
+
+  public constructor() {
+    this.listeners = []
+  }
+
+  public listen(listener: AppEventListener) {
+    this.listeners.push(listener)
+
+    return () => this.listeners.filter(x => x !== listener)
```

Comment:
This doesn't unsubscribe the listener – it only returns a new instance.

Reply:
That's very true!